compiler/src/main/java/com/hadisatrio/libs/android/viewmodelprovider/internal/ConstructorParameter.java

```diff
@@ -0,0 +1,48 @@
+/*
+ *    Copyright (C) 2017 Hadi Satrio
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.hadisatrio.libs.android.viewmodelprovider.internal;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import javax.lang.model.type.TypeMirror;
+
+public class ConstructorParameter {
```

Comment:
I thought an own class for this would be good, since `Pairs` of `Pairs` are hard to read :grin:.